src/node/blockstorage.cpp

```diff
@@ -153,31 +158,54 @@ void BlockManager::FindFilesToPruneManual(std::set<int>& setFilesToPrune, int nM
         return;
     }
 
-    // last block to prune is the lesser of (user-specified height, MIN_BLOCKS_TO_KEEP from the tip)
-    unsigned int nLastBlockWeCanPrune = std::min((unsigned)nManualPruneHeight, chain_tip_height - MIN_BLOCKS_TO_KEEP);
+    unsigned int min_block_to_prune;
+    unsigned int nLastBlockWeCanPrune;
+
+    std::tie(min_block_to_prune, nLastBlockWeCanPrune) = chainman.GetPruneRange(
+        chain, chain_tip_height);
+
+    nLastBlockWeCanPrune = std::min(nLastBlockWeCanPrune, (unsigned)nManualPruneHeight);
```

Comment:
In commit "validation: pruning for multiple chainstates" (b9ac4f7925e2d9dec67be06fdc2a381532581136)

Code would be simpler if this line were dropped and `nManualPruneHeight` was passed as the parameter to `GetPruneRange` above instead of `chain_tip_height`. Passing `chain_tip_height` above has no effect because `GetPruneRange` already takes the min with the `chain.m_chain.Height()`.